src/quest-tree.tsx

```diff
@@ -0,0 +1,97 @@
+import { quests } from "./data";
+import { MissionName, Part } from "./types";
+
+export class QuestForest {
+  private constructor(
+    private roots: QuestTree[]
+  ) {
+  }
+
+  /** Builds a quest forest from the {@link quests} data in data.tsx */
+  static new(): QuestForest {
+    const questsCopy = quests.slice();
```

Comment:
.slice() will only make a shallow copy, you probably want to use JSON.parse(JSON.stringify(quests)) for a deep copy. If you only need a shallow copy, I'd use [...quests] as it's more umm... semantic.

Reply:
A deep copy would be most appropriate here, so that even an accidental added mutation wouldn't cause any side effects on the rest of the app.

We didn't even really need _any_ copy here, since this function does _not_ mutate the data, but I think it's smart to at the very least detach from the original data reference here especially since `quest` is the literal source of truth. And this here at least builds the idea that 'hey, this data is important enough to make a copy, let's make sure we don't mess with it'

Personally I think `.slice()` is an excellent & clean way to make a shallow copy, but I have no quarrels with using a spread operator here.

```suggestion
    const questsCopy = [ ...quests ];
```

This might be good enough to make a deep copy for this data lol
```suggestion
    const questsCopy = JSON.parse(JSON.stringify(quests)) as Quest[];
```

Reply:
I may be misinterpreting something, but, to clarify, spread makes a shallow copy. The stringify way is the only way to deep copy an object (with no functions) that works well enough on old browsers. There's a new function you can use for deep copying but it doesn't have legacy support and I also can't remember what it's called.

Reply:
Yeah, spread and `slice()` are essentially the same thing here. 

I switched it over to the deep copy, seems to work fine. We might run into some trouble in the future if the data in `quests` becomes a bit more complex (e.g. using any sort of class instance to house data).

Reply:
let's just not let that happen, haha